engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/selectModulesScreen/AdvancedGameSetupScreen.java

```diff
@@ -0,0 +1,701 @@
+import org.terasology.engine.SimpleUri;
+import org.terasology.engine.TerasologyConstants;
+import org.terasology.engine.module.DependencyResolutionFailedException;
+import org.terasology.engine.module.ModuleInstaller;
+import org.terasology.engine.module.ModuleManager;
+import org.terasology.engine.module.StandardModuleExtension;
+import org.terasology.i18n.TranslationSystem;
+import org.terasology.math.geom.Vector2i;
+import org.terasology.module.DependencyInfo;
+import org.terasology.module.DependencyResolver;
+import org.terasology.module.Module;
+import org.terasology.module.ModuleMetadata;
+import org.terasology.module.ResolutionResult;
+import org.terasology.naming.Name;
+import org.terasology.registry.In;
+import org.terasology.rendering.nui.Canvas;
+import org.terasology.rendering.nui.CoreScreenLayer;
+import org.terasology.rendering.nui.WidgetUtil;
+import org.terasology.rendering.nui.animation.MenuAnimationSystems;
+import org.terasology.rendering.nui.databinding.Binding;
+import org.terasology.rendering.nui.databinding.ReadOnlyBinding;
+import org.terasology.rendering.nui.itemRendering.AbstractItemRenderer;
+import org.terasology.rendering.nui.layers.mainMenu.ConfirmPopup;
+import org.terasology.rendering.nui.layers.mainMenu.MessagePopup;
+import org.terasology.rendering.nui.layers.mainMenu.UniverseSetupScreen;
+import org.terasology.rendering.nui.layers.mainMenu.WaitPopup;
+import org.terasology.rendering.nui.widgets.ResettableUIText;
+import org.terasology.rendering.nui.widgets.TextChangeEventListener;
+import org.terasology.rendering.nui.widgets.UIButton;
+import org.terasology.rendering.nui.widgets.UICheckbox;
+import org.terasology.rendering.nui.widgets.UILabel;
+import org.terasology.rendering.nui.widgets.UIList;
+import org.terasology.rendering.nui.widgets.UIText;
+import org.terasology.utilities.random.FastRandom;
+import org.terasology.world.generator.internal.WorldGeneratorManager;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.stream.Collectors;
+
+public class AdvancedGameSetupScreen extends CoreScreenLayer {
+
+    public static final ResourceUrn ASSET_URI = new ResourceUrn("engine:advancedGameSetupScreen");
+
+    private static final Logger logger = LoggerFactory.getLogger(SelectModulesScreen.class);
+    private final Comparator<? super ModuleSelectionInfo> moduleInfoComparator = (o1, o2) -> o1.getMetadata()
+            .getDisplayName().toString().compareTo(o2.getMetadata().getDisplayName().toString());
+    @In
+    private ModuleManager moduleManager;
+    @In
+    private Config config;
+
+    private SelectModulesConfig selectModulesConfig;
```

Comment:
This looks slightly out of order without an `@In` yet in the midst of other `@In`ed thing. Could move it down to the private grouping